### PR TITLE
🔨 Make targetSetup into a setupHook

### LINKED ISSUE #305: Target Setup use setuphook
Change target setup to use a setup hook and to be a shell input.

### PR DESCRIPTION
**Pull request by simonrainerson**
mkTargetSetup creates a setupHook which adds calls in the shellHook of
the derivation in which the targetSetup was added as an input. This way
targetSetups are more modular and don't have to be called explicitly.
Fixes #305

### CODE REVIEW

**Comment by abbec** [comment]
It should add itself to the shellHook so we get rid of the attribute on the drv

File: shells.nix [hunk 1]
```diff
@@ -97,7 +98,7 @@ in
                       fi
                       echo ⛑ Changing dir to \"$componentDir\"
                       cd "$componentDir"
-                      ${if drv ? targetSetup then "${drv.targetSetup}/bin/target-setup" else ""}
+                      ${if drv ? targetSetup then "checkTargetSetupMarkers" else ""}
```
Comment (abbec) [inline shells.nix:101]: This should not be needed here. The setup hook should add itself as a shell hook.

**Review by abbec — CHANGES_REQUESTED** [review]

File: shells.nix (hunk 1 above)
Comment (simonrainerson) [reply shells.nix:101]: hookception, good idea.

File: shells.nix [hunk 2]
```diff
@@ -58,7 +58,8 @@ in
                       nativeBuildInputs = oldAttrs.nativeBuildInputs or [ ]
                       ++ oldAttrs.passthru.shellInputs or [ ]
                       ++ oldAttrs.shellInputs or [ ]
-                      ++ [ shellCommands ];
+                      ++ [ shellCommands ]
+                      ++ lib.optional (oldAttrs ? targetSetup) oldAttrs.targetSetup;
```
Comment (abbec) [inline shells.nix:62]: Is there a point (except backwards compatibility) to keep it as a separate drv attribute? Adding it to shellInputs is more intuitive.

Comment (simonrainerson) [reply shells.nix:62]: Backwards compatibility the only reason. But yes, I agree, for next major version we should probably change that. And I guess we could just make a patch release of where main is now and do it in this PR. I don't think we have any known bugs we want to get fixed before doing something that would require a lot of change.

**Review by abbec — APPROVED** [review]

**Pull request merged**